Jetpack Plans: Fix discount message overlapping term filter in some languages

elliottprogrammer opened the pull request:
### Changes proposed in this Pull Request
Fixes discount message overlapping term filter in some languages:
![Markup 2020-10-23 at 13 57 51](https://user-images.githubusercontent.com/11078128/97053846-85ad2180-1538-11eb-9d5c-d7142002104c.png)

### Testing instructions
1. First do a regression test by going to wordpress.com (not localhost) and set your Calypso interface language to French.
    - Go to [https://wordpress.com/me/account](https://wordpress.com/me/account), and change the **Interface Language** to `Français du Canada` (_Français du Canada_ is a good testing example) and be sure to click "Save Account Settings" button.
    - Next go to https://wordpress.com/plans/:site and verify you can see the discount message overlapping problem (Like in the screenshot above)
2. Next run this PR (both Calypso blue and Cloud green)
4. Go to calypso.localhost:3000/plans/:site (Calypso interface should still be set to _Français du Canada_) and verify that the discount message overlapping problem is no longer present and the plans filter bar looks and behaves as expected.
5. Also check [jetpack.cloud.localhost:3001/fr/pricing](http://jetpack.cloud.localhost:3001/fr/pricing) (notice the `/fr` in the url, the page should be showing _French_ language), to verify that the discount message overlapping problem is no longer present.
6. Go back to [https://wordpress.com/me/account](https://wordpress.com/me/account), and change the **Interface Language** back to `English` (if that's you're preferred language).

Should look like the screenshots below:
![Screenshot on 2020-10-23 at 14-47-54](https://user-images.githubusercontent.com/11078128/97056987-20a8fa00-153f-11eb-8e80-404aa8b197bc.png)
![Screenshot on 2020-10-23 at 14-50-03](https://user-images.githubusercontent.com/11078128/97057003-27d00800-153f-11eb-87a2-83070bf8ef39.png)


Fixes: 1196341175636977-as-1198209888368925/f

elliottprogrammer commented on client/my-sites/plans-v2/plans-filter-bar/style.scss:
```diff
@@ -75,7 +75,7 @@
 		margin-left: 8px;
 		margin-right: 16px;
-		flex: 0 0 170px;
+		flex: 0 0 200px;
```
I adjusted this to `flex-basis: 0 0 200px`, because I think this is a better average starting width required by most languages.

monsieur-z reviewed (changes requested):
It works well in Calypso. There's just a little padding needed.
<img width="559" alt="Screen Shot 2020-10-26 at 11 40 39 AM" src="https://user-images.githubusercontent.com/1620183/97194321-656fa380-1780-11eb-9820-d90c354da3d7.png">

When I visit `/fr/pricing` both locally and in prod, I see the page in English. Am I missing something?

elliottprogrammer commented:
> It works well in Calypso. There's just a little padding needed.
> <img alt="Screen Shot 2020-10-26 at 11 40 39 AM" width="559" src="https://user-images.githubusercontent.com/1620183/97194321-656fa380-1780-11eb-9820-d90c354da3d7.png">
> 
> When I visit `/fr/pricing` both locally and in prod, I see the page in English. Am I missing something?

Hmm, I just noticed.. It looks like the filter bar doesn't switch language until you scroll down a little (local and prod), when the component goes "sticky".  (Looks like the component needs a re-render when the language is changed)  Yes, this needs to be fixed.  I'll fix it. 👍

elliottprogrammer commented:
@monsieur-z, now since [ab2c382](https://github.com/Automattic/wp-calypso/pull/46749/commits/ab2c38255691878e1f880fc682f0833a754d4943), visiting the plansFilterBar should now be rendering different languages when visiting `/:locale/pricing`, so for example, visiting `/fr/pricing` should render the plansFilterBar in French.  Also I added padding as suggested.  👍

monsieur-z commented:
Jetpack cloud works well now!

I was mentioning padding at the very right of the bar, sorry if I wasn't clear.
![screenshot](https://user-images.githubusercontent.com/1620183/97306746-44b55580-1835-11eb-859e-f6ee0caaced1.png)

elliottprogrammer commented:
> I was mentioning padding at the very right of the bar, sorry if I wasn't clear.

Ahh, yes now I see.  Nice catch, thank you. Ok I added 15px padding to the right of the discount message, and also adjusted padding on the left of the filter-buttons to 15px, to match.

monsieur-z reviewed (approved):
LGTM! 🚢
